Jenkinsfile

```diff
@@ -0,0 +1,43 @@
+pipeline {
+  agent {
+    node {
+      label 'mac-os'
+    }
+    
+  }
+  stages {
+    stage('Build and test') {
+      when {
+        expression {
+          result = sh (script: "git log -1 | grep '.*\\[ci skip\\].*'", returnStatus: true) // Check if commit message contains skip ci label
+          result != 0 // Evaluate the result
+        }
+        
+      }
+      environment {
+        GIT = credentials('65325e52-5ec0-46a7-a937-f81f545f3c1b')
+      }
+      steps {
+        deleteDir()
```

Comment:
Are you sure this cleanup is necessary? I think every build starts clean automatically...

Reply:
We have removed it now.
I had same doubt while putting this deleteDir() statment. But, we observed the that workspace is not cleaned up on every build in Mac slave.
so we added it as we have the artifacts files created during the build which are the cause of checkout failure. Now, we have modified plugin configuration to checkout the source to matching local branch and added the contents of Build output to gitignore. After this there is no need of this command.